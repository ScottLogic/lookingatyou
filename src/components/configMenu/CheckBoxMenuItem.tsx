/* tslint:disable:jsx-no-lambda */

import React from 'react';
export default function CheckBoxMenuItem(props: {
    name: string;
    onInputChange: (checked: boolean) => void;
    checked: boolean;
}) {
    return (
        <div>
            <label>{props.name}</label>
            <input
                type="checkbox"
                checked={props.checked}
                onChange={event => {
                    props.onInputChange(event.target.checked);
                }}
            />
        </div>
    );
}
