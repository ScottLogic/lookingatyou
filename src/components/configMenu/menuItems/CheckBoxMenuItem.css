.accept {
    position: absolute;
    right: 0%;
}

button {
    align-self: right;
}

.decline,
.accept {
    margin: 15px !important;
    float: right;
}
