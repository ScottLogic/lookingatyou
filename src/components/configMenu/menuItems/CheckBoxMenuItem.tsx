import React from 'react';
import { HelpWith } from '../Help';

export interface ICheckBoxMenuItemProps {
    name: string;
    onInputChange: (checked: boolean) => void;
    checked: boolean;
    helpWith: HelpWith;
}

const CheckBoxMenuItem = React.memo(
    (props: ICheckBoxMenuItemProps) => {
        function onChange(event: React.ChangeEvent<HTMLInputElement>) {
            props.onInputChange(event.target.checked);
        }
        return (
            <p data-tip={true} data-for={HelpWith[props.helpWith]}>
                <div>
                    <label>{props.name}</label>
                    <input
                        type="checkbox"
                        checked={props.checked}
                        onChange={onChange}
                    />
                </div>
            </p>
        );
    },
    (previous, next) =>
        previous.checked === next.checked && previous.name === next.name,
);
export default CheckBoxMenuItem;
