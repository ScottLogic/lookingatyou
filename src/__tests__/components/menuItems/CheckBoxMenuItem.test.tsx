import { mount, shallow } from 'enzyme';
import React from 'react';
import CheckBoxMenuItem, {
    ICheckBoxMenuItemProps,
} from '../../../components/configMenu/menuItems/CheckBoxMenuItem';

let props: ICheckBoxMenuItemProps;
let mockOnInputChange: jest.Mock;

describe('TextBoxMenuItem tests', () => {
    beforeEach(() => {
        mockOnInputChange = jest.fn();
        props = {
            name: 'test',
            checked: false,
            onInputChange: mockOnInputChange,
        };
    });

    it('should render correctly', () => {
        const wrapper = shallow(<CheckBoxMenuItem {...props} />).debug();
        expect(wrapper).toMatchSnapshot();
    });

    it('should call onInputChange on change event', () => {
        const wrapper = mount(<CheckBoxMenuItem {...props} />);
        wrapper.find('input').simulate('change', { target: { value: true } });
        expect(mockOnInputChange).toHaveBeenCalledTimes(1);
    });
});
