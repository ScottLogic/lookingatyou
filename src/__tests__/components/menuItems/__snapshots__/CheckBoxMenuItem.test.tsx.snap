// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBoxMenuItem tests should render correctly 1`] = `
"<p data-tip={true} data-for={[undefined]}>
  <div>
    <label>
      test
    </label>
    <input type=\\"checkbox\\" checked={false} onChange={[Function: onChange]} />
  </div>
</p>"
`;
